### ISSUE TITLE
Wifi off doesn't trigger errors

### ISSUE THREAD

**Issue by saschatimme** [saschatimme/elm-phoenix#32]
Pulled out from  #27 
@bigardone:
>The issue I'm having is that once the connection is made and the channel is joined, if I turn off the wifi it is not receiving any error or anything (not even the onAbnormalClose), until it reaches the socket timeout (which is set to 45 seconds). What I have done is creating a new out port which it is called as the window.navigator.onLine event handlers (https://developer.mozilla.org/en-US/docs/Web/API/NavigatorOnLine/onLine), so it detects it immediately. Does it make sense?

**Comment by saschatimme** [comment]
@bigardone I think an error should be thrown for this scenario.
To find out what is (not) happening I think you should start [here](https://github.com/saschatimme/elm-phoenix/blob/master/src/Phoenix.elm#L484).

This action should be triggered from the underlying websocket package. Is there a message after you turn off the wifi? Is there an error code?  ([Here](https://developer.mozilla.org/en-US/docs/Web/API/CloseEvent) is a list of the error codes).
There is also a `Socket.withDebug` method which should give you more information.

If `Die` is not triggered, you could dig deeper into the underlying websocket package. The relevant interaction with it happens [here](https://github.com/saschatimme/elm-phoenix/blob/master/src/Phoenix.elm#L1055).

Hope this helps you, but if your current solution is working reliable I completely understand if you don't do anything ☺️

**Comment by bigardone** [comment]
@saschatimme I totally agree on that. I'm going to investigate a bit more and share here what I discover in the process :)
